src/storage/store/block.rs

```diff
@@ -66,7 +61,8 @@ fn get_block_page_by_prefix(
         }
 
         Ok(false) // Continue iterating
-    })?;
+    })
+    .map_err(|e| BlockStorageError::TooManyBlocksInResult)?; // TODO: Return the right error
```

Comment:
We need to think about how to handle errors. Existing hub code is using HubError everywhere. This is causing some pain when we integrate. Maybe we should stick with HubError everywhere for now and then revisit later? This HubError is used all the way to the clients, so if we switch to something else, we still need to map to a HubError for backwards compatibility.